cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -44,6 +44,7 @@ var cloudproviders = []string{
 	"aws",
 	"azure",
 	"cloudstack",
+	"external",
```

Comment:
See discussion in https://github.com/kubernetes/kubernetes/pull/47152#issuecomment-310849642